bigquery/storage/managedwriter/adapt/protoconversion.go

```diff
@@ -286,46 +287,72 @@ func storageSchemaToDescriptorInternal(inSchema *storagepb.TableSchema, scope st
 //
 // Messages are always nullable, and repeated fields are as well.
 func tableFieldSchemaToFieldDescriptorProto(field *storagepb.TableFieldSchema, idx int32, scope string, useProto3 bool) (*descriptorpb.FieldDescriptorProto, error) {
+
 	name := strings.ToLower(field.GetName())
+	var fdp *descriptorpb.FieldDescriptorProto
+
 	if field.GetType() == storagepb.TableFieldSchema_STRUCT {
-		return &descriptorpb.FieldDescriptorProto{
+		fdp = &descriptorpb.FieldDescriptorProto{
 			Name:     proto.String(name),
 			Number:   proto.Int32(idx),
 			TypeName: proto.String(scope),
 			Label:    convertModeToLabel(field.GetMode(), useProto3),
-		}, nil
-	}
-
-	// For (REQUIRED||REPEATED) fields for proto3, or all cases for proto2, we can use the expected scalar types.
-	if field.GetMode() != storagepb.TableFieldSchema_NULLABLE || !useProto3 {
-		outType := bqTypeToFieldTypeMap[field.GetType()]
-		fdp := &descriptorpb.FieldDescriptorProto{
-			Name:   proto.String(name),
-			Number: proto.Int32(idx),
-			Type:   outType.Enum(),
-			Label:  convertModeToLabel(field.GetMode(), useProto3),
 		}
-		// Special case: proto2 repeated fields may benefit from using packed annotation.
-		if field.GetMode() == storagepb.TableFieldSchema_REPEATED && !useProto3 {
-			for _, v := range packedTypes {
-				if outType == v {
-					fdp.Options = &descriptorpb.FieldOptions{
-						Packed: proto.Bool(true),
+	} else {
+		// For (REQUIRED||REPEATED) fields for proto3, or all cases for proto2, we can use the expected scalar types.
+		if field.GetMode() != storagepb.TableFieldSchema_NULLABLE || !useProto3 {
+			outType := bqTypeToFieldTypeMap[field.GetType()]
+			fdp = &descriptorpb.FieldDescriptorProto{
+				Name:   proto.String(name),
+				Number: proto.Int32(idx),
+				Type:   outType.Enum(),
+				Label:  convertModeToLabel(field.GetMode(), useProto3),
+			}
+
+			// Special case: proto2 repeated fields may benefit from using packed annotation.
+			if field.GetMode() == storagepb.TableFieldSchema_REPEATED && !useProto3 {
+				for _, v := range packedTypes {
+					if outType == v {
+						fdp.Options = &descriptorpb.FieldOptions{
+							Packed: proto.Bool(true),
+						}
+						break
 					}
-					break
 				}
 			}
+		} else {
+			// For NULLABLE proto3 fields, use a wrapper type.
+			fdp = &descriptorpb.FieldDescriptorProto{
```

Comment:
Why not use proto3_optional instead?

Reply:
Can revisit that decision, but will do it in the scope of a different PR. This code is unmodified but shows as delta due to indentation level.